operators/pkg/instance-controller/controller_test.go

```diff
@@ -12,290 +12,408 @@
-			Spec:   v1.NamespaceSpec{},
-			Status: v1.NamespaceStatus{},
+	BeforeEach(func() {
+		createTenant = true
+		createWorkspace = true
+		createTemplate = true
+	})
+
+	JustBeforeEach(func() {
+		ns = corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testName, Labels: whiteListMap}}
+		webdavSecret = corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Name: webdavSecretName, Namespace: testName},
+			Data:       map[string][]byte{"username": []byte(testName), "password": []byte(testName)},
 		}
-		instanceNs = v1.Namespace{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: InstanceNamespace,
-				Labels: map[string]string{
-					"production": "true",
-					"test-suite": "true",
-				},
+		tenant = clv1alpha1.Tenant{
+			ObjectMeta: metav1.ObjectMeta{Name: testName},
+			Spec: clv1alpha1.TenantSpec{
+				Email: "test@email.me",
 			},
-			Spec:   v1.NamespaceSpec{},
-			Status: v1.NamespaceStatus{},
 		}
-		webdavSecret = v1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "webdav-secret",
-				Namespace: InstanceNamespace,
+		workspace = clv1alpha1.Workspace{ObjectMeta: metav1.ObjectMeta{Name: testName}}
+		template = clv1alpha2.Template{
+			ObjectMeta: metav1.ObjectMeta{Name: testName, Namespace: testName},
+			Spec: clv1alpha2.TemplateSpec{
+				WorkspaceRef:    clv1alpha2.GenericRef{Name: testName},
+				EnvironmentList: []clv1alpha2.Environment{environment},
 			},
-			Data: map[string][]byte{
-				"username": []byte("username"),
-				"password": []byte("password"),
-			},
-			StringData: nil,
-			Type:       "",
 		}
-		instance = crownlabsv1alpha2.Instance{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      InstanceName,
-				Namespace: InstanceNamespace,
-			},
-			Spec: crownlabsv1alpha2.InstanceSpec{
-				Running: true,
-				Template: crownlabsv1alpha2.GenericRef{
-					Name:      TemplateName,
-					Namespace: TemplateNamespace,
-				},
-				Tenant: crownlabsv1alpha2.GenericRef{
-					Name: TenantName,
-				},
+		instance = clv1alpha2.Instance{
+			ObjectMeta: metav1.ObjectMeta{Name: testName, Namespace: testName},
+			Spec: clv1alpha2.InstanceSpec{
+				Template: clv1alpha2.GenericRef{Name: testName, Namespace: testName},
+				Tenant:   clv1alpha2.GenericRef{Name: testName, Namespace: testName},
+				Running:  runInstance,
 			},
-			Status: crownlabsv1alpha2.InstanceStatus{},
 		}
-		instance2 = crownlabsv1alpha2.Instance{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      InstanceName + "persistent",
-				Namespace: InstanceNamespace,
-			},
-			Spec: crownlabsv1alpha2.InstanceSpec{
-				Running: true,
-				Template: crownlabsv1alpha2.GenericRef{
-					Name:      TemplateName + "persistent",
-					Namespace: TemplateNamespace,
-				},
-				Tenant: crownlabsv1alpha2.GenericRef{
-					Name: TenantName,
-				},
-			},
-			Status: crownlabsv1alpha2.InstanceStatus{},
+
+		Expect(k8sClient.Create(ctx, &ns)).ToNot(HaveOccurred())
```

Comment:
```suggestion
		Expect(k8sClient.Create(ctx, &ns)).To(Succeed())
```

Nit. Just a bit more idiomatic (same for all other occurrences)